### DISCUSSION TITLE
href in modelview

### DISCUSSION THREAD

**Discussion by sheldygg** [aminalaee/sqladmin#633] [Q&A] [answered]
How i can create href in modelview?
I have a Telegram username and I want to open the link in a separate tab when clicked.

```python
class UserModel(DefaultModelView, model=User):
    column_formatters = {
        User.username: lambda user, a: f"\n<a href='t.me/{user.username}'>{user.username}</a>\n" if user.username else ""
    }
```

I did this but I get a simple line =(

**Comment by aminalaee** [comment, marked as answer]
This works:

```py
from markupsafe import Markup

class UserAdmin(ModelView, model=User):
    column_formatters = {
        "username": lambda user, u: Markup(f"<a href='https://t.me/{user.id}'>{user.id}</a>" if user.id else "#")
    }
```

Searching the similar solution for flask-admin usually gives the answer: https://stackoverflow.com/questions/37258668/flask-admin-how-to-change-formatting-of-columns-get-urls-to-display
